src/lib/rng/hmac_drbg/hmac_drbg.cpp

```diff
@@ -20,6 +20,11 @@ HMAC_DRBG::HMAC_DRBG(std::unique_ptr<MessageAuthenticationCode> prf,
    {
    BOTAN_ASSERT_NONNULL(m_mac);
 
+   if(reseed_interval == 0 || reseed_interval > static_cast<size_t>(1) << 48)
```

Comment:
This overflows and causes problems on 32-bit builds.

We could either omit the second check when sizeof(size_t) == 4 (since it is not possible to actually express a reseed interval exceeding the requirement in that case), or else reduce the limit below the SP mandated maximum. Which seems preferable to me since 2**48 is a *lot* of output. Maybe drop our max to 2**24 or so??

You forgot to update the constructor taking just an `EntropySource&`. I think it would be best to hoist these checks up to an anon namespace function to validate both reseed_interval and max bytes / request parameters so the logic is not duplicated 3 times.

Reply:
I totally forgot about 32 bit platforms. 2^24 seems OK to me, I also documented that in the API doc. Introduced a new function `check_limits()` to deduplicate code as you suggested, all in 91f925164.